clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -171,6 +195,7 @@ public void removeSensor(String name) {
                     }
                 }
             }
```

Comment:
All parents get the `record` from children, but what about the other way around? If a parent gets a record, then it does not update the children. Should the condition be that for a parent sensor to be removed, then all its children should also not have received updates? IOW a parent sensor can never be removed. i.e., children would have to get expired first before it can be removed. Does that make sense?